core/src/main/res/values-night/color.xml

```diff
@@ -13,4 +13,8 @@
   <color name="color_on_primary">@color/white</color>
   <color name="color_on_surface">@color/white</color>
   <color name="color_on_error">@color/white</color>
+
+  <color name="text_primary">#000000</color>
```

Comment:
these should reference colors with names so here `@color/black` obviously but the others can reference the `grey700`/`grey500` you originally had. This is all about lending semantic meaning to our color system. 

I link [this video](https://www.youtube.com/watch?v=Owkf8DhAOSo) to a lot of people but it is just really good at explaining the whole terrible situation that exists.

Reply:
@macgills The reason why I used those terminologies is that if I define
gray700 as #9e9e9e 
gray500 as #616161 in colors.xml (day)
then I have to define corresponding inverted colors as 
gray500 as #9e9e9e 
gray700 as #616161 in colors.xml (night) which is misleading and confusing.

So, I think it would be good to have three text color levels for using in respective places. Also, we should not use directly black or any color, instead have some `text_primary` defined in color so that in case we have to change the `text_primary` color we just have to change in one place and not look for text color in everyplace to change.

Reply:
in your main colors file have 
```
grey_500
grey_700

example_color=grey_500
```
in night colors
```
example_color= grey_700
```

The definition of actual colors doesn't change because grey_500 should always be grey_500, example_color is the one that changes.

This allows grey_500 to be any number of colors, it is a layer of abstraction.